airtime_mvc/application/common/TaskManager.php

```diff
@@ -0,0 +1,212 @@
+<?php
+
+/**
+ * Class TaskManager
+ */
+final class TaskManager {
+
+    /**
+     * @var array tasks to be run
+     */
+    protected $_taskList = [
+        AirtimeTask::UPGRADE,  // Always run the upgrade first
+        AirtimeTask::CELERY
+    ];
+
+    /**
+     * @var TaskManager singleton instance object
+     */
+    protected static $_instance;
+
+    /**
+     * @var int TASK_INTERVAL_SECONDS how often, in seconds, to run the TaskManager tasks,
+     *                                if they need to be run
+     */
+    const TASK_INTERVAL_SECONDS = 60;
+
+    /**
+     * @var $con PDO Propel connection object
+     */
+    private $_con;
+
+    /**
+     * Private constructor so class is uninstantiable
+     */
+    private function __construct() {
+    }
+
+    /**
+     * Get the singleton instance of this class
+     *
+     * @return TaskManager the TaskManager instance
+     */
+    public static function getInstance() {
+        if (!self::$_instance) {
+            self::$_instance = new TaskManager();
+        }
+        return self::$_instance;
+    }
+
+    /**
+     * Run all tasks that need to be run
+     */
+    public function runTasks() {
+        // If there is data in auth storage, this could be a user request
+        // so we should lock the TaskManager to avoid blocking
+        if ($this->_isUserSessionRequest()) return;
```

Comment:
A user request is any request that has a Zend_Auth object stored in session. This is meant to prevent blocking on user-navigation through Airtime. We send a lot of anonymous background requests that are still bootstrapped, which is good because we still want to trigger the tasks when the user is not logged in.